contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java

```diff
@@ -287,14 +235,41 @@ public void build() {
                 conf.setReducerClass(AvroStoreBuilderReducer.class);
             }
 
-            logger.info("Number of chunks: " + numChunks + ", number of reducers: " + numReducers
-                        + ", save keys: " + saveKeys + ", reducerPerBucket: " + reducerPerBucket
-                        + ", buildPrimaryReplicasOnly: " + buildPrimaryReplicasOnly);
             logger.info("Building store...");
-            RunningJob job = JobClient.runJob(conf);
 
-            // Once the job has completed log the counter
-            Counters counters = job.getCounters();
+            // The snipped below copied and adapted from: JobClient.runJob(conf);
+            // We have more control in the error handling this way.
+
+            JobClient jc = new JobClient(conf);
+            RunningJob runningJob = jc.submitJob(conf);
+            Counters counters = runningJob.getCounters();
+
+            try {
+                if (!jc.monitorAndPrintJob(conf, runningJob)) {
+                    // For some datasets, the number of chunks that we calculated is inadequate.
+                    // Here, we try to identify if this is the case.
+                    long mapOutputBytes = counters.getCounter(Task.Counter.MAP_OUTPUT_BYTES);
+                    int numChunks = conf.getInt(AbstractStoreBuilderConfigurable.NUM_CHUNKS, 1);
+                    long averageNumberOfBytesPerChunk = mapOutputBytes / numChunks / cluster.getNumberOfPartitions();
+                    if (averageNumberOfBytesPerChunk > HadoopStoreWriter.MAX_CHUNK_SIZE) {
+                        logger.warn("The number of bytes per chunk is too high. averageNumberOfBytesPerChunk = " +
+                                averageNumberOfBytesPerChunk + ". Consider using a smaller chunk size. " +
+                                "The MapReduce job will be retried one more time with adjusted settings.");
+                        setNumChunksAndNumReduceTasks(mapOutputBytes, conf);
+                        try {
+                            RunningJob newJob = JobClient.runJob(conf);
+                            counters = newJob.getCounters();
+                        } catch (IOException e) {
+                            throw new VoldemortException("Even with adjusted settings, BnP's MapReduce job failed.", e);
+                        }
+                    } else {
+                        logger.error("Job Failed: " + runningJob.getFailureInfo());
+                        throw new VoldemortException("BnP's MapReduce job failed.");
+                    }
+                }
+            } catch (InterruptedException ie) {
+                Thread.currentThread().interrupt();
```

Comment:
Why is an interruptedException caught here, if throwing the interruptedException from here fails the code, does it need to be handled ?